package.py

```diff
@@ -0,0 +1,911 @@
+            'aarch64',
+            'arm64',
+        ]
+
+        self.oses = [
+            'Darwin',
+            'Windows',
+            'Linux',
+        ]
+
+        self.arch = platform.machine()
+        self.os = platform.system()
+        self.target_arch = os.environ.get('MACOS_BUILD_ARCH', self.arch)
+
+        if self.is_mac():
+            os.environ.setdefault('ONEFILE', '0')
+
+        python_bin = 'python3'
+        if self.is_windows():
+            python_bin = 'python'
+
+        self.rotki_version = subprocess.check_output(
+            f'{python_bin} setup.py --version',
+            shell=True,
+            encoding='utf8',
+        ).strip()
+
+        if os.environ.get('ROTKI_VERSION') is None:
+            os.environ.setdefault('ROTKI_VERSION', self.rotki_version)
+
+    @staticmethod
+    def check_repo() -> None:
+        output = subprocess.check_output(
+            'git branch --show-current',
+            encoding='utf8',
+            shell=True,
+        ).strip()
+
+        if output != 'master':
+            logger.info(f'Current branch is {output}, skipping check')
+            return
+
+        unmerged_commits = subprocess.check_output(
+            'git rev-list HEAD..bugfixes --no-merges | wc -l | xargs echo -n',
+            encoding='utf8',
+            shell=True,
+        ).strip()
+
+        if unmerged_commits != "0":
+            logger.error(
+                f"Found {unmerged_commits} in bugfixes that have not been merged for release",
+            )
+            exit(1)
+        logger.info("branch is up to date with bugfixes")
+
+    def check_environment(self) -> None:
+        if self.arch not in self.archs:
+            logger.error(f'{self.arch} is not supported, packaging only supports {self.archs}')
+            exit(1)
+
+        if self.os not in self.oses:
+            logger.error(
+                f'{platform.system()} is not supported, packaging only supports {self.oses}',
+            )
+            exit(1)
+
+        if not self.is_ci() and not os.environ.get('VIRTUAL_ENV'):
+            logger.error('The script should not run outside a virtual environment if not on CI')
+            exit(1)
+
+        self.check_repo()
+
+    def is_mac(self) -> bool:
+        return self.os == 'Darwin'
+
+    def is_linux(self) -> bool:
+        return self.os == 'Linux'
+
+    def is_windows(self) -> bool:
+        return self.os == 'Windows'
+
+    def is_mac_runner(self) -> bool:
+        return self.is_mac() and self.is_ci()
+
+    def is_universal2(self) -> bool:
+        return self.is_mac_runner() or os.environ.get('MACOS_BUILD_ARCH') == 'universal2'
+
+    def is_x86_64(self) -> bool:
+        return self.arch in ['x86_64', 'AMD64']
+
+    def os_tag(self) -> str:
+        if self.is_mac():
+            return 'macos'
+        if self.is_linux():
+            return 'linux'
+        if self.is_windows():
+            return 'windows.exe'
+        raise Exception(f'Invalid os {self.os}')
```

Comment:
is the `.exe` extension here intended?

Reply:
The window filename ends with `windows.exe` the macos and linux binaries on the other hand have no extensions so it was either this or including the extension where that tag is consumed. This is certainly easier to do.

Reply:
Add a comment for that as I would also have the same question on why it's here. So I guess someone reading it later will also wonder why the difference.

Reply:
I renamed to backend suffix btw to make it a bit more clear